book/src/prefabs/prefabs_technical_explanation.md

```diff
@@ -0,0 +1,323 @@
+    );
+
+    type Result = Handle<A>;
+
+    fn add_to_entity(
+        &self,
+        entity: Entity,
+        system_data: &mut Self::SystemData,
+        _: &[Entity],
+    ) -> Result<Handle<A>, SpecsError> {
+        let handle = match *self {
+            AssetPrefab::Handle(ref handle) => handle.clone(),
+            AssetPrefab::File(ref name, ref format, ref options) => system_data.0.load(
+                name.as_ref(),
+                format.clone(),
+                options.clone(),
+                (),
+                &system_data.2,
+            ),
+        };
+        system_data.1.insert(entity, handle.clone()).map(|_| handle)
+    }
+
+    fn load_sub_assets(
+        &mut self,
+        progress: &mut ProgressCounter,
+        system_data: &mut Self::SystemData,
+    ) -> Result<bool, SpecsError> {
+        let handle = match *self {
+            AssetPrefab::File(ref name, ref format, ref options) => Some(system_data.0.load(
+                name.as_ref(),
+                format.clone(),
+                options.clone(),
+                progress,
+                &system_data.2,
+            )),
+            _ => None,
+        };
+        if let Some(handle) = handle {
+            *self = AssetPrefab::Handle(handle);
+        }
+        Ok(true)
+    }
+}
+```
+
+So, there are two main differences to this `PrefabData` compared the `Transform` example.
+The first difference is that the `add_to_entity` function now return a `Handle<A>`.
+The second difference is that `load_sub_assets` is implemented, this is because we load
+a sub asset. The `load_sub_assets` function here will do the actual loading, and morph the
+internal representation to the `AssetPrefab::Handle` variant, so when `add_to_entity` runs later
+it will straight up use the internally stored `Handle`.
+
+### Special `PrefabData` implementations
+
+There are a few special blanket implementations provided by the asset system:
+
+* `Option<T>` for all `T: PrefabData`.
+* Tuples of types that implemented `PrefabData`, up to a size of 20.
+
+### Deriving `PrefabData` implementations
+
+Amethyst supplies a derive macro for creating the `PrefabData` implementation for the following scenarios:
+
+* Single `Component`
+* Aggregate `PrefabData` structs which contain other `PrefabData` constructs, and optionally simple data `Component`s
+
+In addition, deriving a `Prefab` requires that `amethyst::ecs::Entity` and
+ `amethyst:assets::{PrefabData, PrefabError, ProgressCounter}` are imported
+ and visible in the current scope. This is due to how Rust macros work.
+
+An example of a single `Component` derive:
+
+```rust,no_run,noplaypen
+# #[macro_use] extern crate amethyst;
+# #[macro_use] extern crate serde_derive;
+# use amethyst::assets::{Asset, AssetStorage, Loader, Format, Handle, ProgressCounter, PrefabData, PrefabError};
+# use amethyst::ecs::{WriteStorage, ReadExpect, Read, Entity, error::Error as SpecsError, DenseVecStorage, Component};
```

Comment:
I suggest to not hide imports of `Prefab` and `amethyst_assets::{PrefabData, PrefabError, ProgressCounter}` since this should not be forgotten.